homeassistant/components/ecobee/switch.py

```diff
@@ -31,25 +31,22 @@ async def async_setup_entry(
     """Set up the ecobee thermostat switch entity."""
     data: EcobeeData = hass.data[DOMAIN]
 
-    async_add_entities(
-        [
-            EcobeeVentilator20MinSwitch(
-                data,
-                index,
-                (await dt_util.async_get_time_zone(thermostat["location"]["timeZone"]))
-                or dt_util.get_default_time_zone(),
-            )
-            for index, thermostat in enumerate(data.ecobee.thermostats)
-            if thermostat["settings"]["ventilatorType"] != "none"
-        ],
-        update_before_add=True,
-    )
-
-    async_add_entities(
-        EcobeeSwitchAuxHeatOnly(data, index)
+    entities: list[SwitchEntity] = [
+        EcobeeVentilator20MinSwitch(
+            data,
+            index,
+            (await dt_util.async_get_time_zone(thermostat["location"]["timeZone"]))
+            or dt_util.get_default_time_zone(),
+        )
         for index, thermostat in enumerate(data.ecobee.thermostats)
-        if thermostat["settings"]["hasHeatPump"]
-    )
+        if thermostat["settings"]["ventilatorType"] != "none"
+    ]
+
+    for index, thermostat in enumerate(data.ecobee.thermostats):
+        if thermostat["settings"]["hasHeatPump"]:
+            entities.append(EcobeeSwitchAuxHeatOnly(data, index))
```

Comment:
This one can also be an comprehension but using `entities.extend(..)`

Reply:
https://github.com/home-assistant/core/pull/129917/commits/76c3e82b83a93807de70e69fb5f679c5c3d3512b